shared/node.go

```diff
@@ -0,0 +1,116 @@
+package shared
+
+import (
+	"fmt"
+	"log"
+	"pocket/consensus"
+	"pocket/persistence"
+	"pocket/pre_p2p"
+	"pocket/shared/config"
+	"pocket/shared/crypto"
+	"pocket/shared/types"
+	"pocket/utility"
+	"time"
+
+	"pocket/shared/modules"
+)
+
+// TODO: SHould we create an interface for this as well?
+type Node struct {
+	modules.Module
```

Comment:
Just wanted to confirm that we wanted to do inheritance here and not interface enforcement...

Reply:
```
Node{
		Module:    nil, // doesn't need to be here...
		pocketBus: nil,
		Address:   "",
	}
```

Reply:
Done.